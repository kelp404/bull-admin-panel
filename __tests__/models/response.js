const Response = require('../../lib/models/response');

afterEach(() => jest.restoreAllMocks());

test('initial response with request id and ws', () => {
  const response = new Response({
    requestId: 'id',
    ws: {id: 1}
  });

  expect(response).toMatchSnapshot();
});

test('call end() will set isDidResponse as true', () => {
  const response = new Response({
    requestId: Math.random().toString(36).substr(2),
    ws: {}
  });

  expect(response.isDidResponse).toBeFalsy();
  response.end();
  expect(response.isDidResponse).toBeTruthy();
});

test('can not send data twice', () => {
  const data = {url: 'http://example.com'};
  const response = new Response({
    requestId: Math.random().toString(36).substr(2),
    ws: {
      send: jest.fn()
    }
  });

  response.json(data);
  response.json(data);
  expect(response.ws.send).toBeCalledTimes(1);
});

test('send success json response', () => {
  const data = {url: 'http://example.com'};
  const response = new Response({
    requestId: 'lb11ge3aa0f',
    ws: {
      send: jest.fn(message => expect(message).toMatchSnapshot())
    }
  });

  jest.spyOn(response, 'end');

  response.json(data);
  expect(response.ws.send).toBeCalled();
  expect(response.end).toBeCalled();
});

test('send failed json response', () => {
  const data = {url: 'http://example.com'};
  const response = new Response({
    requestId: 'lb11ge3aa0f',
    ws: {
      send: jest.fn(message => expect(message).toMatchSnapshot())
    }
  });

  jest.spyOn(response, 'end');

  response.json(data, 500);
  expect(response.ws.send).toBeCalled();
  expect(response.end).toBeCalled();
});
