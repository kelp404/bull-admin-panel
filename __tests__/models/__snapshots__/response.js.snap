// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial response with request id and ws 1`] = `
Response {
  "isDidResponse": false,
  "requestId": "id",
  "ws": Object {
    "id": 1,
  },
}
`;

exports[`send failed json response 1`] = `"{\\"type\\":\\"response\\",\\"id\\":\\"lb11ge3aa0f\\",\\"status\\":500,\\"body\\":{\\"url\\":\\"http://example.com\\"}}"`;

exports[`send success json response 1`] = `"{\\"type\\":\\"response\\",\\"id\\":\\"lb11ge3aa0f\\",\\"status\\":200,\\"body\\":{\\"url\\":\\"http://example.com\\"}}"`;
